hollow/src/main/java/com/netflix/hollow/core/read/engine/object/HollowObjectTypeReadStateShard.java

```diff
@@ -144,6 +144,9 @@ public long readLong(int ordinal, int fieldIndex) {
     }
 
     public Boolean readBoolean(int ordinal, int fieldIndex) {
+        if (ordinal < 0)
```

Comment:
Not in fact long and int works pretty well.
I've tried to run tests in order to crash, but without any result. 
Probably this scenario is the most tested.

Reply:
curious - what do we return for long and int? sentinel values?

Reply:
My bad, I've made the same mistake as previous developer - values go to different buckets. There's still the issue with readLong, etc... Here it's a bit more tricky due to the fact, that method uses primitives, but ordinal still might be -1.

Reply:
What about `if (ordinal < 0) return Long.MIN_VALUE;` in order to convert to a sentinel value?
The best option would be to return null, but that would mean changing method signature to use Long / Integer / Double , which would lead to quite a big change (and definitely to some side effects).

Reply:
I see, there's something like `DefaultMissingDataHandler.java`, which already have all this stuff.

Reply:
I've came up to the same decision. I'll open another PR, closing this one.